### ISSUE TITLE
Consulta em postgress com acento.

### ISSUE THREAD

**Issue by thyagobrejao** [laravelbrasil/forum#74]
Bom dia povo..

Estou usando postgres mas estou tento dificuldade com select quando o campo tem acentos...
Estou usando assim para buscar
```
$municipios = Municipio::where(DB::raw('unaccent(nome)'), 'ilike', '%'.$request->get('q').'%')
```
Porém ele retorna erro
```
7 ERROR: function unaccent(character varying) does not exist
```
Executando a mesma query direto no banco, ela funciona... para isso eu executei apenas uma vez o comando  **CREATE EXTENSION IF NOT EXISTS unaccent** direto no banco
Tomei por base esse artigo
http://www.codegeo.com.br/2013/04/ignorando-acentos-em-consultas-com-php.html
O que está faltando??
Obrigado.

**Comment by IgorDePaula** [comment]
O nome do campo tem acento ou o valor do campo tem acento?

Bom...a mensagem de erro diz que tal função nao existe. Verifique a versão de seu postgre.

**Comment by thyagobrejao** [comment]
@IgorDePaula O valor do campo que tem acento.

No banco está salvo **Brasília** **com** acento por exemplo, mas quando eu faço um select por **Brasilia** **sem** acento, aí não retorna anada..

**Comment by maykonmendel** [comment]
@thyagobrejao 

Por que não tenta utilizar uma consulta utilizando o like com o caractere coringa?

**Comment by thyagobrejao** [comment]
@maykonmendel Como faria isso?? Não sobrecarregaria o banco??

**Comment by maykonmendel** [comment]
@thyagobrejao 

Não, pois é uma função muito utilizada para se trazer exatamente o resultado esperado. Vou te dar um exemplo. Eu tenho uma tabela de nomes e na coluna nome_completo tenho armazenado o nome Luís. Tem acento, correto? Pode tentar a consulta SELECT * FROM nome WHERE nome_completo LIKE '%Luís%'.

Caso queira entender melhor, te indico esse link http://www.devmedia.com.br/sql-server-2005-operadores-like-e-not-like/17292.

**Comment by thyagobrejao** [comment]
A tá,, mas isso eu já faço..usando ilike para tratar maiúsculo e minúsculo..
Com o MySQL é mais fácil resolver, só alterar a colation.. Mas no postgres que estou apanhando com esses acentos!!!

Esqueci de responder ao @IgorDePaula, minhas versão do posrgres é a mesma do Homestead,,acredito que seja a última né!!

**Comment by maykonmendel** [comment]
Não trabalho muito com postgres, então não posso me aprofundar mais, mas se for com MySQL e SQL Server, estou a disposição.

**Comment by paulofreitas** [comment]
Eu não costumo usar o PostgreSQL, mas eu consegui usar a função `unaccent()` normalmente por aqui. Verifique se você rodou o comando `CREATE EXTENSION unaccent` no mesmo *schema* que está usando na conexão do Laravel. :wink:

Inicialmente eu fiz um teste sem instalar a extensão `unaccent`, o erro foi exatamente o mesmo. Então rodei o `psql` e executei o comando `CREATE EXTENSION unaccent` no mesmo *schema* que uso no Laravel (`public`). Você também pode especificar o *schema* diretamente no DDL: `CREATE EXTENSION unaccent SCHEMA public`
Depois que fiz isso eu pude usar a função `unaccent()` normalmente no `DB::raw()`. 👍
